### PR TITLE
Python/WinRT Assorted Fixes

### PR DESCRIPTION
**Pull request by devhawk**
* Manage PyObject* lifetimes with RAII wrappers
* add dealloc functions for structs and winrt_base
* rework native __await__ function for IAsync* types to better match original Python implementation
* fix array view construction for pass_array parameters
* rework how root module name is accessed in the projection generation code 
* add module name to projected types
* Update python tests to include generated folder in search path
* Remove explicit init apartment calls from python tests
* remove explicit debug flags (/Zi and /Od) from setup.py

Example generated code available: https://github.com/devhawk/pywinrt-output/pull/6/files

### CODE REVIEW

File: src/tool/python/strings/pybase.h [hunk 1]
```diff
@@ -750,36 +750,52 @@ namespace py
         }
     };
 
+    struct pyobj_ptr_traits
+    {
+        using type = PyObject*;
```
Comment (Scottj1s) [inline src/tool/python/strings/pybase.h]: >PyObject* [](start = 21, length = 9)

consider typedef-ing a PyObjectPtr and using it throughout this header - everywhere you pass and return raw PyObject* pointers, to make things exception safe.

**Comment by devhawk** [comment]
fixes #191

**Comment by devhawk** [comment]
fixes #104

File: src/tool/python/strings/pybase.h (hunk 1 above)
Comment (devhawk) [reply src/tool/python/strings/pybase.h]: Done. Also cleaned up the way WinRT delegates and async complete callback capture and clean up python state

**Comment by devhawk** [comment]
https://github.com/devhawk/pywinrt-output/pull/6/files updated w/ latest projected code

**Pull request merged**